controllers/users.js

```diff
@@ -291,7 +291,19 @@ const updateSelf = async (req, res) => {
       await userQuery.setIncompleteUserDetails(userId);
     }
 
-    const user = await userQuery.addOrUpdate(req.body, userId);
+    let updatedUserData = req.body;
+    if (req.body.discordId) {
+      // The request is from /verify, set in_discord to true
+      updatedUserData = {
+        ...updatedUserData,
+        roles: {
+          ...req.userData.roles,
+          in_discord: true,
+        },
+      };
+    }
+
+    const user = await userQuery.addOrUpdate(updatedUserData, userId);
```

Comment:
When the user verifies and links their discord account, a `PATCH` call is made to `users/self` ([Ref.](https://github.com/Real-Dev-Squad/website-my/blob/develop/app/controllers/discord.js#L20-L52)).
When linking their discord account to their RDS account, we are setting the `in_discord` role to `true`.
This will ensure that when we add the discord account details to the user's data upon successful verification, we also set their `in_discord` role to `true`.